bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BUnionType.java

```diff
@@ -189,11 +205,15 @@ public boolean isNilable() {
     }
 
     private boolean checkNillable(List<BType> memberTypes) {
+        this.resolving = true;
         for (BType t : memberTypes) {
             if (t.isNilable()) {
+                this.resolving = false;
                 return true;
             }
+            this.resolving = false;
```

Comment:
Is this correct?  Doesn't this need to be true until all members are iterated over?